rc_django/views/rest_proxy.py

```diff
@@ -0,0 +1,230 @@
+        service_name = service
+        if service == 'iasystem':
+            service_name = 'iasystem_uw'
+
+        proxy = RestProxy(service_name)
+        response = proxy.get_api_response(url, headers)
+
+        # logger.debug("get_api_response url: {}, status: {}, data: {}".format(
+        #    url, response.status, response.data))
+
+        use_pre = True if (service == "calendar") else False
+        is_image = False
+
+        if (response.status == 200 and
+                re.match(r'/idcard/v1/photo/[0-9A-F]{32}', url)):
+            # Handle known images
+            is_image = True
+            content = b64encode(response.data)
+        elif use_pre:
+            content = response.data
+        else:
+            content = proxy.formatted
+
+        context.update({
+            "url": unquote(url),
+            "content": content,
+            "json_data": proxy.json,
+            "response_code": response.status,
+            "time_taken": "{:f} seconds".format(proxy.duration),
+            "headers": response.headers,
+            "override_user": user_service.get_override_user(),
+            "use_pre": use_pre,
+            "is_image": is_image,
+        })
+
+        try:
+            loader.get_template("restclients/extra_info.html")
+            context["has_extra_template"] = True
+            context["extra_template"] = "restclients/extra_info.html"
+        except TemplateDoesNotExist:
+            pass
+
+        try:
+            search_template_path = re.sub(r"[.?].*$", "", url)
+            search_template = "proxy/{}{}.html".format(service,
+                                                       search_template_path)
+            loader.get_template(search_template)
+            context["search_template"] = search_template
+            context["search"] = format_search_params(url)
+        except TemplateDoesNotExist:
+            context["search_template"] = None
+
+        return context
+
+    def get(self, request, *args, **kwargs):
+        """
+        The proxy view for API URLs.
+        """
+        # Using args for these URLs for backwards-compatibility
+        service = args[0]
+        url = args[1] if len(args) > 1 else ""
+
+        if service == "sws" or service == "gws":
+            kwargs["actas_user"] = True
+
+        if request.GET:
+            try:
+                url = "{}?{}".format(url, urlencode(request.GET))
+            except UnicodeEncodeError as err:
+                return HttpResponse(
+                    'Bad URL param given to the restclients browser')
+
+        kwargs["service"] = service
+        kwargs["url"] = url
+        try:
+            context = self.get_context_data(**kwargs)
+        except (AttributeError, ImportError):
+            return HttpResponse("Missing service: {}".format(service),
+                                status=404)
+
+        return self.render_to_response(context)
+
+    def post(self, request, *args, **kwargs):
```

Comment:
The post method belongs to RestSearchView and is missing the api call part

Reply:
Fixed, the custom forms now use an action that point to the rest proxy view.